types/node/package.json

```diff
@@ -3,5 +3,8 @@
     "types": "index",
     "typesVersions": {
         "<=4.8": { "*": ["ts4.8/*"] }
+    },
+    "dependencies": {
+        "undici-types": "~5.25.1"
```

Comment:
This is the first version of undici-types available, but it reflects a newer minor version than is actually included in Node.js so far. It sounded from https://github.com/nodejs/undici/issues/2261#issuecomment-1728599811 like we will be able to wait on a new Node.js version comes out containing undici 5.25 and bump the `@types/node` version accordingly if we want this dependency mapping to be perfectly accurate. cc @Ethan-Arrowood

Reply:
Yeah, I don't think there's much of an issue shipping this now and then getting synced up later. If there's an actual issue (some bug or whatever), then just tell me exactly which version you want me to publish and I'll manually go a publish the `undici-types` version.

Reply:
JFI, the latest undici version for Node.js versions:
- Node.js 18 - [`5.22.1`](https://github.com/nodejs/node/blob/v18.x/src/undici_version.h)
- Node.js 20 - [`5.25.2`](https://github.com/nodejs/node/blob/v20.x/src/undici_version.h)

Reply:
Kinda concerned about the impact of this on users of `@types/node` I've already seen a ton of duplicate installations of the package with different versions, using a restrictive version range for undici is likely going to make this worse.

Reply:
Can you elaborate? This isn’t undici, just undici-types.

Reply:
All those patch versions are identical, minus tiny bugfixes for other issues.

Reply:
More curious if we should make this `>=5.22.1` to allow users squash/consolidate their dependencies more easily, a lot of libraries currently depend on `@types/node@*` to avoid this already, some don't and it causes multiple instances of @types/node to be installed which sometimes have been conflicting.

I'm not sure if I'm overstating the issue of if this is simply not a concern.

Reply:
Yeah, mainly I assume this is a question of "what if there are multiple types packages in a project", where previously `@types/node` had no deps so could only "conflict" with itself in a compatible fashion, but adding a dep expands that risk (where restricting the version range increases that risk). I'm not sure that it will actually cause a problem, though.

Reply:
The downside to loosening the range is indicating to a Node.js 20.3.X user (for example) that they have access to undici features that they actually don’t. undici just recently added support for HTTP2, so using a `>=` range could have incorrectly led users of a Node.js that hadn’t adopted that undici version to try to access those features, had this PR happened earlier.

On the other hand, I think you’re just automatically screwed if you have two `@types/node` versions in the same program, even without dependencies.